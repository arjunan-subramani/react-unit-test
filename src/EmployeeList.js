import React, { Component } from 'react';
import ReactTable from "react-table";
import "react-table/react-table.css";

export default class EmployeeList extends Component {
  render() {
    return (
      <div className="employeelist">
        <ReactTable
          showPagination={false}
          minRows={6}
          data={this.props.data}  
          columns={this.props.columns}  
        />
      </div>
    )
  }
}
