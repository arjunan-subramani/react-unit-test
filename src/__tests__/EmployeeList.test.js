import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import EmployeeList from '../EmployeeList';

describe("EmployeeList Component", () => {
	const data = [
  	{ id: 'INC001', name: 'John', role: 'CEO' }
	];
	const columns = [
    { Header: 'Employee ID', accessor: 'id' },
    { Header: 'Name', accessor: 'name' },
    { Header: 'Designation', accessor: 'role' }
  ];
  
	it('should shallow render component without crashing', () => {
    shallow(<EmployeeList />);
	});

	it('should full render component without crashing', () => {
    mount(<EmployeeList data={data} columns={columns} />);
	});

	it('should static render component without crashing', () => {
    render(<EmployeeList data={data} columns={columns} />);
	});

	it('should render the component initial layout', () => {	
		const tree = renderer.create(<EmployeeList data={data} columns={columns} />);
  	expect(tree.toJSON()).toMatchSnapshot();
	});
});
