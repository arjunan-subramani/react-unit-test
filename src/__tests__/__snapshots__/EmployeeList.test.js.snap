// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeList Component should render the component initial layout 1`] = `
<div
  className="employeelist"
>
  <div
    className="ReactTable"
    style={Object {}}
  >
    <div
      className="rt-table"
      role="grid"
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "300px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
        >
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Employee ID
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Name
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header -cursor-pointer"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              Designation
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "300px",
          }
        }
      >
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -odd"
            role="row"
          >
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              INC001
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              John
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              CEO
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          >
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
            <div
              className="rt-td"
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="-loading"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
